chore: add GitHub Pages

ericcrosson-bitgo opened the pull request:
To demo the site locally, run:

```
npm ci
npm --prefix packages/website start
```

This commit also includes a GitHub Actions workflow to continuously
deploy the website artifacts to GitHub Pages.

This is a _heavy_ WIP but I'd also like to get this into a
continously-deploying state quickly, to prevent it from rotting too
long internally.

I plan to contribute regularly to these docs, so regardless of the
starting state we should have a good trajectory.

bitgopatmcl reviewed (changes requested):
They're not super large images, but just to keep bloat down while possible, I think the images in `website/docs/how-to-guides/img/` aren't referenced anywhere.

ericcrosson-bitgo commented:
> They're not super large images, but just to keep bloat down while possible, I think the images in `website/docs/how-to-guides/img/` aren't referenced anywhere.

I removed the `website/docs/how-to-guides/img` directory and rebased on master

bitgopatmcl reviewed (approved):
🚀
